test/new-e2e/language-detection/node_test.go

```diff
@@ -0,0 +1,47 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/stretchr/testify/require"
+)
+
+const nodeProg = `
+const http = require('http');
+
+const hostname = '127.0.0.1';
+const port = 3000;
+
+const server = http.createServer((req, res) => {
+  res.statusCode = 200;
+  res.setHeader('Content-Type', 'text/plain');
+  res.end('Hello World');
+});
+
+server.listen(port, hostname, () => {
+  console.log('Server running');
+});
+`
+
+func (s *languageDetectionSuite) installNode() {
+	s.Env().VM.Execute(
+		"curl -fsSL https://deb.nodesource.com/setup_20.x | sudo -E bash - && " +
+			"sudo apt-get install -y nodejs")
+	nodeVersion := s.Env().VM.Execute("node --version")
+	require.True(s.T(), strings.HasPrefix(nodeVersion, "v20."))
+}
+
+func (s *languageDetectionSuite) TestNodeDetection() {
+	s.installNode()
+
+	s.Env().VM.Execute(fmt.Sprintf(`echo "%s" > prog.js`, nodeProg))
+	s.Env().VM.Execute("nohup node prog.js >myscript.log 2>&1 </dev/null &")
```

Comment:
I actually did try it from a goroutine, but unfortunately when I tested from my local machine the VM seemed to be single threaded, i.e. I had to immediately sleep on the main thread in order for the goroutine to execute. I think it would be solved by `assert.Eventually()`, although that means the first run is always going to fail.

Reply:
Interesting, this is fine, the only limitation here is that we keep the process running after the end of a subtest. We will add a `ExecuteInBackground` function to the VM, so getting context on what you tested. We might return a pid for it so that it can be killed before exiting the subtest.

Reply:
I can add a kill statement at the end of the subtest, actually maybe that would be better. Right now each test is independent so it doesn't matter that the process continues to be run, but that's not a guarantee 😜

Reply:
Also, returning a pid would be really nice for my use case, since I need to identify it using the pid in the final output 🙂